### PR TITLE
Add HasLinkMixin

### PR DESCRIPTION
**Pull request by AidanCurrah**
This mixin deals with the saving of the fields and also adds some nice helper functions for access the data. I've left sections alone and some of the links in the header.

### CODE REVIEW

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/components/admin.py [hunk 1]
```diff
@@ -7,3 +7,8 @@
 @admin.register(CallToAction)
 class CallToActionAdmin(admin.ModelAdmin{% if cookiecutter.sections == 'yes' %}, UsedOnAdminMixin{% endif %}):
     list_display = ['title'{% if cookiecutter.sections == 'yes' %}, 'pages_used_on'{% endif %}]
+    fieldsets = (
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/components/admin.py:10]: Not that it makes much difference, but FYI you can this here instead:

```python
fields = ['kicker', 'title', 'link_text', 'link_page', 'link_url']
```

Comment (AidanCurrah) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/components/admin.py:10]: It's just my preference to have `fieldsets` in `admin.py`s and `fields` in `forms.py`s

Comment (lewiscollard) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/components/admin.py:10]: I'm OK with that. :)

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/utils/models.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    class Meta:
+        abstract = True
+
+    def clean(self):
+        if self.link_text and (not self.link_page and not self.link_url):
+            raise ValidationError({
+                'link_page': 'Please supply either a "link page" or "link URL"',
+            })
+
+    @property
```
Comment (joe-hastings) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/utils/models.py:40]: make this a `@cached_property`

**Review by joe-hastings — APPROVED** [review]

**Pull request merged**